pkg/appgw/health_probes.go

```diff
@@ -107,11 +107,18 @@ func (c *appGwConfigBuilder) generateHealthProbe(backendID backendIdentifier) *n
 		probe.Path = to.StringPtr(backendID.Path.Path)
 	}
 
-	// backend protocol "https" now supports to use certificate signed by well-known root certificate as well as trusted self-signed certificate
-	if protocol, _ := annotations.BackendProtocol(backendID.Ingress); protocol == annotations.HTTPS {
+	if backendID.Backend.ServicePort.IntVal == 443 {
```

Comment:
can you we use `backendID.Backend.ServicePort.String()` ?
Since this is a union type, it can get int or string as input. Will IntVal work when using string: "443"

Reply:
> can you we use `backendID.Backend.ServicePort.String()` ?
> Since this is a union type, it can get int or string as input. Will IntVal work when using string: "443"

Updated.